ompi/runtime/ompi_rte.c

```diff
@@ -569,6 +569,23 @@ int ompi_rte_init(int *pargc, char ***pargv)
         /* if we get PMIX_ERR_UNREACH indicating that we cannot reach the
          * server, then we assume we are operating as a singleton */
         if (PMIX_ERR_UNREACH == ret) {
+            /* if we are in a PMI or SLURM environment with two tasks or more,
+             * we probably do not want to start singletons */
+            char *size_str = getenv("PMI_SIZE");
+            if (NULL == size_str) {
+                size_str = getenv("SLURM_NPROCS");
+            }
+            int size = (NULL != size_str)?atoi(size_str):1;
+            if (1 < size) {
+                char *rank_str = getenv("PMI_RANK");
+                if (NULL == rank_str) {
+                    rank_str = getenv("SLURM_PROCID");
+                }
+                int rank = (NULL != rank_str)?atoi(rank_str):0;
+                if (0 == rank) {
```

Comment:
What if this really is rank 0? Should -1 be used instead?

Reply:
The rationale for being `0` is we do our best to limit the warning message to rank `0`. But if we cannot figure out the rank (likely caused by a busted environment?) I'd rather have all the ranks print the warning message than none.

Reply:
I agree with @ggouaillardet on this one.